static/src/javascripts/projects/common/modules/experiments/ab-test-targeting.js

```diff
@@ -0,0 +1,39 @@
+define([
+    'common/utils/config',
+], function(
+    config
+) {
+    function clean(string) {
+        return string.trim().toLowerCase();
+    }
+
+    /**
+     * Get the campaigns for a test.
+     *
+     * @param test      an A/B test object (with an 'id' property)
+     * @return {Array}  campaigns associated with this test
+     */
+    function campaignsFor(test) {
+        try {
+            return config.page.campaigns.filter(function (campaign) {
+                return campaign.fields && clean(campaign.fields.campaignId) === clean(test.id);
```

Comment:
:guardswoman: Fair doos. Sounds like something to bring up with targeting so we can be sure of the format of campaign IDs because free text sounds like a world of pain